### ISSUE TITLE
Return all reports of a test failure

### ISSUE THREAD

**Issue by cfjedimaster** [lirantal/swagger-lint-api#3]
## Expected Behavior

I've got a custom rule that looks for quote characters. It works fine, but only reports the first result in a JSON file. It should return all, right?

## Current Behavior

It returns just the first match.

**Comment by lirantal** [comment]
Probably should return all as you would expect from something like eslint too. Indeed it currently just halts on first attempt through all of them. Need to fix that.

**Comment by cfjedimaster** [comment]
Some notes that may not apply to you - but I wanted to take a stab at solving this too. I began by trying out to recursively grab all the paths in a deeply nested object such that it would match our key. I totally failed, asked for help on Twitter, and got some interesting solutions.

https://twitter.com/cirsca/status/1151931114463756288

https://twitter.com/jakerella/status/1151931514168455168

https://runkit.com/jwcarroll/find-all-properties-recursively

**Comment by lirantal** [comment]
For sure, if you're planning on submitting a PR that would be! 💥🤘

**Comment by cfjedimaster** [comment]
Option 2 doesn't work.

**Comment by cfjedimaster** [comment]
What's your feeling on option 3 and then using eval to check the strings? Yes I know eval is Bad(tm), but it could be a quick and dirty solution.

**Comment by cfjedimaster** [comment]
Oh nice, we can use lodash and this: https://lodash.com/docs/4.17.14#get

**Comment by cfjedimaster** [comment]
I've got a prototype of this working. Kinda. It returns an array of results with valid:false in each which is more info then we need, instead we should return {value:false, reasons: ....}. But it's something. Sharing via a Gist for now. I'm using an internal function inside a static method in the class which feels wrong. 

https://gist.github.com/cfjedimaster/101ab6431fb0a1de012348d51cd691fc

**Comment by lirantal** [comment]
Yeah not a fan of using eval ;-)

- I'm good with lodash
- As for the private function, just declare it as another static function for the class, that would be fine and we can add some unit tests to test that in isolation

How about you roll this into a PR?

**Comment by lirantal** [comment]
Also, great progress! 🤗❤️

**Comment by cfjedimaster** [comment]
So I tried adding it as another static method and it didn't work. See this:

https://gist.github.com/cfjedimaster/09fc16cd3eafb41d2a9ec913e785dad5

When this is executed, I get an error saying findProps is not defined. Like the one static method can't call the other.

**Comment by cfjedimaster** [comment]
Nevermind - for one static method to call another, you just add `this`.

**Comment by cfjedimaster** [comment]
I'd like to file a PR, but it breaks the test. Is there a way to commit and ignore the tests?

**Comment by lirantal** [comment]
@cfjedimaster ad d `--no-verify` at the end of the git commit/push to skip validation on the client side and push the repo to GitHub.

**Comment by cfjedimaster** [comment]
Done - thank you.

**Comment by cfjedimaster** [comment]
Query - it feels wrong - but can I also update my PR to add support for returning a message. So imagine this in SchemaUtils.js:

```js
      const ret = validationSchema.keyValidator(value)

      if (!ret) {
        results.push({
          valid: false,
          path: m
        })
      }
```

I'd modify  `if(!ret)` to support seeing if ret is an object, and if it's an object we require a valid key and look for a 'message' key that gets added to the push. I'd still support boolean results for backwards compat (and simplicity sake).

**Comment by lirantal** [comment]
@cfjedimaster yep, that makes sense. Let's make sure we update the README to explain that though.
